### PR TITLE
added ref to Patient and RelatedPerson to collection.collector

### PR DESCRIPTION
**Pull request by ryma2fhir**
>The FHIR R5 snapshot of the [Specimen](https://hl7.org/fhir/5.0.0-snapshot3/Specimen.html) resource, has a change to Specimen.collection.collector, adding a reference to Patient and RelatedPerson. 
>
>Backport the R5 functionality by adding a reference the [UKCore-Patient](https://simplifier.net/guide/uk-core-implementation-guide-stu3-sequence/Home/ProfilesandExtensions/Profile-UKCore-Patient?version=1.6.0) and [UKCore-RelatedPerson](https://simplifier.net/guide/uk-core-implementation-guide-stu3-sequence/Home/ProfilesandExtensions/Profile-UKCore-RelatedPerson?version=1.6.0) profiles

### CODE REVIEW

**Comment by ryma2fhir** [comment]
>Error: expect(received).toBeFalsy()

    Received: "WARNING The target profile [https://fhir.hl7.org.uk/StructureDefinition/UKCore-Patient](https://simplifier.net/resolve?fhirVersion=R4&scope=uk.nhsdigital.r4&canonical=https://fhir.hl7.org.uk/StructureDefinition/UKCore-Patient) is not a valid constraint on the base ([CanonicalType](https://simplifier.net/resolve?fhirVersion=R4&scope=uk.nhsdigital.r4&canonical=http://hl7.org/fhir/StructureDefinition/PractitionerRole%5D%5D)[[http://hl7.org/fhir/StructureDefinition/Practitioner],](https://simplifier.net/resolve?fhirVersion=R4&scope=uk.nhsdigital.r4&canonical=http://hl7.org/fhir/StructureDefinition/Practitioner%5D,) CanonicalType[[http://hl7.org/fhir/StructureDefinition/PractitionerRole]] at Specimen.collection.collector [ Location - StructureDefinition]
    WARNING The target profile [https://fhir.hl7.org.uk/StructureDefinition/UKCore-RelatedPerson](https://simplifier.net/resolve?fhirVersion=R4&scope=uk.nhsdigital.r4&canonical=https://fhir.hl7.org.uk/StructureDefinition/UKCore-RelatedPerson) is not a valid constraint on the base ([CanonicalType](https://simplifier.net/resolve?fhirVersion=R4&scope=uk.nhsdigital.r4&canonical=http://hl7.org/fhir/StructureDefinition/PractitionerRole%5D%5D)[[http://hl7.org/fhir/StructureDefinition/Practitioner],](https://simplifier.net/resolve?fhirVersion=R4&scope=uk.nhsdigital.r4&canonical=http://hl7.org/fhir/StructureDefinition/Practitioner%5D,) CanonicalType[[http://hl7.org/fhir/StructureDefinition/PractitionerRole]] at Specimen.collection.collector [ Location - StructureDefinition]"

This is a warning rather than an error. Might need some discussion on this

**Comment by ryma2fhir** [comment]
I'm hoping I have read this correctly, but References are extensible http://hl7.org/fhir/R5/references.html#reference

**Comment by ryma2fhir** [comment]
Created an extension and added to collection.collector. Suggest we come up with a standard naming convention for these R5 backports for future reference (and to allow others to know of breaking changes when updating to R5).

===
TODO
===
collection.collector has a 0..1 cardinality (in R4 & R5). Suggest rules to check extension and vice versa to ensure we cannot have 2 references at the same time

**Comment by ryma2fhir** [comment]
rules added

File: structuredefinitions/UKCore-Specimen.xml [hunk 1]
```diff
@@ -159,6 +169,12 @@
         <targetProfile value="https://fhir.hl7.org.uk/StructureDefinition/UKCore-Practitioner" />
         <targetProfile value="https://fhir.hl7.org.uk/StructureDefinition/UKCore-PractitionerRole" />
       </type>
+      <constraint>
```
Comment (APageNHS) [inline structuredefinitions/UKCore-Specimen.xml:172]: This possibly needs moving up due to the other changes

Comment (APageNHS) [reply structuredefinitions/UKCore-Specimen.xml:172]: Ignore that, its ok where it is, but the human value needs updating

Comment (APageNHS) [reply structuredefinitions/UKCore-Specimen.xml:172]: and the expression?

File: structuredefinitions/Extension-UKCore-SpecimenCollectionCollector.xml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  <type value="Extension" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Extension" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Extension.url">
+      <path value="Extension.url" />
+      <fixedUri value="http://hl7.org/fhir/5.0/StructureDefinition/extension-Specimen.collection.collector" />
+    </element>
+    <element id="Extension.value[x]">
+      <path value="Extension.value[x]" />
```
Comment (APageNHS) [inline structuredefinitions/Extension-UKCore-SpecimenCollectionCollector.xml:38]: This need a min value=1 in lline with design approach

File: structuredefinitions/UKCore-Specimen.xml [hunk 3]
```diff
@@ -143,6 +143,16 @@
...
+    <element id="Specimen.collection.extension:specimenCollectionCollectorR5">
+      <path value="Specimen.collection.extension" />
+      <sliceName value="specimenCollectionCollectorR5" />
+      <min value="0" />
+      <max value="1" />
+      <type>
+        <code value="Extension" />
+        <profile value="http://hl7.org/fhir/5.0/StructureDefinition/extension-Specimen.collection.collector" />
+      </type>
+      <isModifier value="false" />
```
Comment (APageNHS) [inline structuredefinitions/UKCore-Specimen.xml:155]: As discussed, is missing its end of element tag

Comment (APageNHS) [reply structuredefinitions/UKCore-Specimen.xml:155]: And do you want to add a short description to the value[x] in the extension in the profile to force it to render?

File: structuredefinitions/Extension-UKCore-SpecimenCollectionCollector.xml (hunk 2 above)
Comment (ryma2fhir) [reply structuredefinitions/Extension-UKCore-SpecimenCollectionCollector.xml:43]: done

File: structuredefinitions/UKCore-Specimen.xml (hunk 3 above)
Comment (ryma2fhir) [reply structuredefinitions/UKCore-Specimen.xml:161]: done

File: structuredefinitions/UKCore-Specimen.xml (hunk 1 above)
Comment (ryma2fhir) [reply structuredefinitions/UKCore-Specimen.xml]: moved expression to extension and rewrote human value

**Review by APageNHS — APPROVED** [review]

**Pull request merged**